src/TropicalGeometry/groebner_basis.jl

```diff
@@ -0,0 +1,255 @@
+#   T. Markwig, Y. Ren: Computing tropical varieties over fields with valuation
+###
+
+
+
+#=======
+return true if f is homogeneous (w.r.t. total degree)
+return false otherwise
+=======#
+function is_homogeneous(f::Union{AbstractAlgebra.Generic.MPoly{K},fmpq_mpoly,fmpz_mpoly} where {K})
+  leadexpv,tailexpvs = Iterators.peel(exponent_vectors(f))
+  d = sum(leadexpv)
+  for tailexpv in tailexpvs
+    if d!=sum(tailexpv)
+      return false
+    end
+  end
+  return true
+end
+export is_homogeneous
+
+function is_homogeneous(I::MPolyIdeal{K} where {K})
+  # todo: test whether generators are interreduced
+  @warn "is_homogeneous: merely checking whether given generators are homogeneous, can result in false negative"
+
+  for f in gens(I)
+    if !is_homogeneous(f)
+      return false
+    end
+  end
+  return true
+end
+
+
+
+#=======
+tropical Groebner basis
+todo: proper documentation
+Example:
+
+val_2 = ValuationMap(QQ,2)
+Kx,(x,y,z) = PolynomialRing(QQ,3)
+w = [0,0,0]
+I = ideal([x+2*y,y+2*z])
+groebner_basis(I,val_2,w)
+
+Kt,t = RationalFunctionField(QQ,"t")
+val_t = ValuationMap(Kt,t)
+Ktx,(x,y,z) = PolynomialRing(Kt,3)
+w = [0,0,0]
+I = ideal([x+t*y,y+t*z])
+groebner_basis(I,val_t,w,return_lead=true)
+=======#
+@doc Markdown.doc"""
+    groebner_basis(I::Ideal, val::ValuationMap, w::Vector; complete_reduction::Bool, return_lead)
+
+Computes a Groebner basis of `I` over a field with valuation `val` with respect to weight vector `w`, that is a finite generating set of `I` whose initial forms generate the initial ideal with respect to `w`.
+
+For the definitions of initial form, initial ideal and Groebner basis see [Maclagan-Sturmfels, Section 2.4].
+
+# Warning
+`I` must be homogeneous if `val` is non-trivial or `w` contains non-positive entries. If `val` is trivla and `w` contains only non-negative entries, then what is computed is a regular Groebner basis with respect to a weighted ordering with weight vector `w`.
+
+# Examples
+```jldoctest
+julia> Kx,(x0,x1,x2,x3,x4,x5) = PolynomialRing(QQ,6);
+
+julia> Cyclic5Homogenized = ideal([x1+x2+x3+x4+x5,
+                                   x1*x2+x2*x3+x3*x4+x1*x5+x4*x5,
+                                   x1*x2*x3+x2*x3*x4+x1*x2*x5+x1*x4*x5+x3*x4*x5,
+                                   x1*x2*x3*x4+x1*x2*x3*x5+x1*x2*x4*x5+x1*x3*x4*x5+x2*x3*x4*x5,
+                                   -x0^5+x1*x2*x3*x4*x5]);
+
+julia> Katsura5Homogenized = ideal([-x0+x1+2*x2+2*x3+2*x4+2*x5,
+                                    -x0*x1+x1^2+2*x2^2+2*x3^2+2*x4^2+2*x5^2,
+                                    -x0*x2+2*x1*x2+2*x2*x3+2*x3*x4+2*x4*x5,
+                                    x2^2-x0*x3+2*x1*x3+2*x2*x4+2*x3*x5,
+                                    2*x2*x3-x0*x4+2*x1*x4+2*x2*x5]);
+
+julia> val_2 = ValuationMap(QQ,2); # 2-adic valuation
+
+julia> val_3 = ValuationMap(QQ,3); # 3-adic valuation
+
+julia> w = [0,0,0,0,0,0];
+
+julia> groebner_basis(Cyclic5Homogenized, val_2, w, complete_reduction=true)
```

Comment:
@YueRen I wondered about that too. Do we even need to test this in OSCAR? It definitely makes the documentation harder to read. And I would suspect that these things are already tested inside Singular? Otherwise we should move this to be a regular test instead of a jldoctest.

Reply:
I moved these to the tests. @YueRen will have to add new examples.